store/iavlstore.go

```diff
@@ -123,6 +125,55 @@ func (st *iavlStore) ReverseIterator(start, end []byte) Iterator {
 	return newIAVLIterator(st.tree.Tree(), start, end, false)
 }
 
+// Query implements ABCI interface, allows queries
+//
+// by default we will return from (latest height -1),
+// as we will have merkle proofs immediately (header height = data height + 1)
+// If latest-1 is not present, use latest (which must be present)
+// if you care to have the latest data to see a tx results, you must
+// explicitly set the height you want to see
+func (st *iavlStore) Query(req abci.RequestQuery) (res abci.ResponseQuery) {
+	if len(req.Data) == 0 {
+		msg := "Query cannot be zero length"
+		return sdk.ErrTxParse(msg).Result().ToQuery()
+	}
+
+	tree := st.tree
+	height := req.Height
+	if height == 0 {
+		latest := tree.Version64()
+		if tree.VersionExists(latest - 1) {
+			height = latest - 1
+		} else {
+			height = latest
+		}
+	}
+	// store the height we chose in the response
+	res.Height = height
+
+	switch req.Path {
+	case "/store", "/key": // Get by key
+		key := req.Data // Data holds the key bytes
+		res.Key = key
```

Comment:
How about collapsing the above 2 lines into one.

Reply:
i use key a few places, it's nicer to clarify what it is